### ISSUE TITLE
System.InvalidOperationException: The calling thread cannot access this object because a different thread owns it.

### ISSUE THREAD

**Issue by Bluscream** [chocolatey/ChocolateyGUI#914]
```
---------------------------
Unhandled Exception
---------------------------
System.InvalidOperationException: The calling thread cannot access this object because a different thread owns it.

   at System.Windows.Threading.Dispatcher.VerifyAccess()

   at MahApps.Metro.Controls.Dialogs.DialogManager.ShowMessageAsync(MetroWindow window, String title, String message, MessageDialogStyle style, MetroDialogSettings settings) in /_/src/MahApps.Metro/Controls/Dialogs/DialogManager.cs:line 161

   at ChocolateyGui.Common.Windows.Services.DialogService.<ShowMessageAsync>d__8.MoveNext() in C:\projects\chocolateygui\Source\ChocolateyGui.Common.Windows\Services\DialogService.cs:line 43

--- End of stack trace from previous location where exception was thrown ---

   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)

   at ChocolateyGui.Common.Windows.ViewModels.Items.PackageViewModel.<Update>d__181.MoveNext() in C:\projects\chocolateygui\Source\ChocolateyGui.Common.Windows\ViewModels\Items\PackageViewModel.cs:line 570

--- End of stack trace from previous location where exception was thrown ---

   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

   at ChocolateyGui.Common.Windows.ViewModels.LocalSourceViewModel.<UpdateAll>d__69.MoveNext() in C:\projects\chocolateygui\Source\ChocolateyGui.Common.Windows\ViewModels\LocalSourceViewModel.cs:line 229

--- End of stack trace from previous location where exception was thrown ---

   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)

   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

   at System.Windows.Threading.DispatcherOperation.InvokeImpl()

   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)

   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)

   at System.Windows.Threading.DispatcherOperation.Invoke()

   at System.Windows.Threading.Dispatcher.ProcessQueue()

   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)

   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)

   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)

   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)

   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)

   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)

   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)

   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)

   at System.Windows.Application.RunDispatcher(Object ignore)

   at System.Windows.Application.RunInternal(Window window)

   at ChocolateyGui.App.Main(String[] args) in C:\projects\chocolateygui\Source\ChocolateyGui\App.xaml.cs:line 116
---------------------------
OK   
---------------------------
```

![](https://i.imgur.com/52KyV6W.png)

![](https://i.imgur.com/1DIh4DQ.png)

**Comment by Bluscream** [comment]
Maybe it's the same issue as #319

Here's my packages.config: https://gist.github.com/Bluscream/8a0a2b60cb2c95877301852828ef2351#file-packages-config

**Comment by punker76** [comment]
@gep13 I think this should be fixed by #876

**Comment by Bluscream** [comment]
> @Bluscream can you please install 0.20.0 and confirm whether you are still seeing this issue?
> 
> Thanks

I keep my choco up-to-date anyway, and i haven't encountered it so far. will re-open if it is not over ^^
